File: language/javascript/extractor/src/model/coref/file.ts


import * as path from 'path';
import { hashToInt64 } from '../../util';

import { BaseModel } from './base-model';

export interface File extends BaseModel {
  name: string;
  extension: string;
  relativePath: string;
  locationOid?: bigint;
}

export function createFile(relativePath: string): File {
  const name = path.basename(relativePath);
  const extension = path.extname(name);
  const oid = hashToInt64(relativePath);
  return {
    oid,
    name,
    extension,
    relativePath,
  };
}

File: language/properties/extractor/src/main/java/com/alipay/codequery/properties/model/File.java

package com.alipay.codequery.properties.model;


public class File extends Node{
  /**
   * File's relative path.
   */
  public String relativePath;
  /**
   * File's extention.
   */
  public String extension;
  /**
   * File's name.
   */
  public String name;

  /**
   * Extract corresponding mybatis object.
   * @return
   */
  public com.alipay.codequery.properties.dal.mybatis.domain.File extractFile(){
    return new com.alipay.codequery.properties.dal.mybatis.domain.File(this.oid, this.relativePath, this.extension, this.name);
  }

  /**
   * Constructor.
   * @param oid
   */
  public File(Long oid) {
    super(oid);
  }

  /**
   * Default constructor.
   */
  public File() {
  }
}
